src/server/scripts/BrokenIsles/zone_orderhall_warrior.cpp

```diff
@@ -0,0 +1,885 @@
+    PHASE_DANICA    = 5090
+};
+
+enum Quests
+{
+    QUEST_ODYN_AND_THE_VALARJAR = 39654,
+};
+
+class npc_danica_the_reclaimer : public CreatureScript
+{
+public:
+    npc_danica_the_reclaimer() : CreatureScript("npc_danica_the_reclaimer") { }
+
+    struct npc_danica_the_reclaimerAI : public ScriptedAI
+    {
+        npc_danica_the_reclaimerAI(Creature* creature) : ScriptedAI(creature) { }
+
+        static uint8 const DanicaPath01Size = 6;
+        Position const DanicaPath01[DanicaPath01Size] =
+        {
+            { 1050.219f, 7232.470f, 100.5846f },
+            { 1046.207f, 7240.372f, 100.5846f },
+            { 1040.963f, 7245.498f, 100.6819f },
+            { 1034.726f, 7250.083f, 100.5846f },
+            { 1027.422f, 7257.835f, 100.5846f },
+            { 1027.542f, 7259.735f, 100.5846f }
+        };
+
+        static uint8 const DanicaPath02Size = 6;
+        Position const DanicaPath02[DanicaPath02Size] =
+        {
+            { 1018.493f, 7247.438f, 100.5846f },
+            { 1013.535f, 7243.327f, 100.5846f },
+            { 1007.063f, 7235.723f, 100.5846f },
+            { 1003.337f, 7229.650f, 100.5846f },
+            { 995.4549f, 7227.286f, 100.5846f },
+            { 984.4410f, 7224.357f, 100.5846f }
+        };
+
+        static uint8 const DanicaPath03Size = 5;
+        Position const DanicaPath03[DanicaPath03Size] =
+        {
+            { 962.5208f, 7223.089f, 100.5846f },
+            { 934.2795f, 7223.116f, 100.5846f },
+            { 911.8507f, 7223.776f, 100.5846f },
+            { 879.0139f, 7224.100f, 100.9079f },
+            { 851.691f, 7224.5490f, 109.5846f }
+        };
+
+        enum Texts
+        {
+            SAY_FIRST_LINE  = 1,
+            SAY_SECOND_LINE = 2,
+            SAY_THIRD_LINE  = 3
+        };
+
+        // Should be the player
+        // Personal spawn ? Demon Creator is the player who accepts the quest, no phasing involved but the quest giver dissapears and gets replaced with a new one
+        void IsSummonedBy(Unit* summoner) override
+        {
+            if (summoner->GetTypeId() != TYPEID_PLAYER)
+                return;
+
+            me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+            _summonerGuid = summoner->GetGUID();
+            _scheduler.Schedule(Seconds(2), [this, summoner](TaskContext /*context*/)
+            {
+                me->GetMotionMaster()->MoveSmoothPath(0, DanicaPath01, DanicaPath01Size, false, true);
+                Talk(SAY_FIRST_LINE, summoner);
+            });
+        }
+
+        void MovementInform(uint32 /*type*/, uint32 id) override
+        {
+            switch (id)
+            {
+                case 0:
+                    _scheduler.Schedule(Seconds(10), [this](TaskContext /*context*/)
+                    {
+                        Player* player = ObjectAccessor::FindConnectedPlayer(_summonerGuid);
+                        me->GetMotionMaster()->MoveSmoothPath(1, DanicaPath02, DanicaPath02Size, false, true);
```

Comment:
No it's not, shouldn't the function only set catmullroll for flying creatures only and smoothgroundpath for non-flying units ?
Which function should I use if I want an NPC to move by path ?
Small exception for danica, she is not using catmullroll while flying, but I can't find a function for it except ```me->GetMotionMaster()->MovePath()``` which requires the path to be in the database.